lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -5441,7 +5441,15 @@ namespace Js
 
     Var JavascriptOperators::OP_InitCachedScope(Var varFunc, const Js::PropertyIdArray *propIds, Field(DynamicType*)* literalType, bool formalsAreLetDecls, ScriptContext *scriptContext)
     {
-        ScriptFunction *func = JavascriptGeneratorFunction::Is(varFunc) || JavascriptAsyncFunction::Is(varFunc) ?
+        bool isGAFunction = JavascriptFunction::Is(varFunc);
+        Assert(isGAFunction);
```

Comment:
Rest of the code has to comply or we have a type confusion bug. I checked the call sites, and they do comply.

My concern is that someone can add another call to this function in the future where varFunc is not guaranteed to be a JavascriptFunction. This was a problem with the code before, but if you are changing these casts, it's better to eliminate that possibility.